lib/zendesk_apps_support/manifest.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+module ZendeskAppsSupport
+  class Manifest
+    class << self
+      def private_attr_reader(*names)
+        private
+        attr_reader(*names)
+      end
+    end
+
+    LEGACY_URI_STUB = '_legacy'
+
+    RUBY_TO_JSON = {
+      name: 'name',
+      requirements_only: 'requirementsOnly',
+      version: 'version',
+      author: 'author',
+      framework_version: 'frameworkVersion',
+      single_install: 'singleInstall',
+      signed_urls: 'signedUrls',
+      no_template: 'noTemplate',
+      default_locale: 'defaultLocale',
+      original_locations: 'location',
+      private: 'private',
+      oauth: 'oauth',
+      original_parameters: 'parameters',
+      domain_whitelist: 'domainWhitelist',
+      remote_installation_url: 'remoteInstallationURL',
+      terms_conditions_url: 'termsConditionsURL',
+      google_analytics_code: 'gaID'
+    }
+
+    attr_reader(*RUBY_TO_JSON.keys)
+
+    alias_method :requirements_only?, :requirements_only
+    alias_method :signed_urls?, :signed_urls
+    alias_method :single_install?, :single_install
+    alias_method :private?, :private
+
+    def no_template?
+      if no_template.is_a?(Array)
+        false
+      else
+        no_template
+      end
+    end
+
+    def no_template_locations
+      no_template || []
+    end
+
+    def location?
+      locations != { 'zendesk' => {} }
```

Comment:
Oh sorry, I actually misread this as `locations ||= { ... }`

In that case, it seems odd that having no location is equated to being exactly equal to the (empty) locations object structure, which might change if we e.g. add more products or change `zendesk` to `support`.

Could we implement this in a way that doesn't have to know anything about defaults? e.g. `locations.values.all?(&:empty)`? That way `{}`, `{'zendesk' => {}}` and `{'support' => {}}` are all location-less.